### ISSUE TITLE
Modeling issue: Let me know if someone is familiar with this.

### ISSUE THREAD

**Issue by soorajsh** [DARKS-ML/Insider-Threat-Detection-using-ML#24]
I was developing a model to deploy that using the explore1.ipynb jupyter file that was on github a month ago. Whle creating that model file, I encountered this warning:

ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.

While searching for the solution I got: 

Normally when an optimization algorithm does not converge, it is usually because the problem is not well-conditioned, perhaps due to a poor scaling of the decision variables. There are a few things you can try.

    Normalize your training data so that the problem hopefully becomes more well conditioned, which in turn can speed up convergence. One possibility is to scale your data to 0 mean, unit standard deviation using Scikit-Learn's StandardScaler for an example. Note that you have to apply the StandardScaler fitted on the training data to the test data.
    Related to 1), make sure the other arguments such as regularization weight, C, is set appropriately.
    Set max_iter to a larger value. The default is 1000.
    Set dual = True if number of features > number of examples and vice versa. This solves the SVM optimization problem using the dual formulation. Thanks @Nino van Hooff for pointing this out, and @JamesKo for spotting my mistake.
    Use a different solver, for e.g., the L-BFGS solver if you are using Logistic Regression. See @5ervant's answer.

Note: One should not ignore this warning.

This warning came about because

    Solving the linear SVM is just solving a quadratic optimization problem. The solver is typically an iterative algorithm that keeps a running estimate of the solution (i.e., the weight and bias for the SVM). It stops running when the solution corresponds to an objective value that is optimal for this convex optimization problem, or when it hits the maximum number of iterations set.

    If the algorithm does not converge, then the current estimate of the SVM's parameters are not guaranteed to be any good, hence the predictions can also be complete garbage.

Let me know if someone could explain this to me. 
Regards,
